    @Test
    public void testLogIoException() throws Exception {
        //I can't put wrong file here. Checkstyle fails before check started.
        //I saw some usage of file or handling of wrong file in Checker, or somewhere
        //in checks running part. So I had to do it with reflection to improve coverage.
        final TranslationCheck check = new TranslationCheck();
        final DefaultConfiguration checkConfig = createModuleConfig(TranslationCheck.class);
        final TestMessageDispatcher dispatcher = new TestMessageDispatcher();
        check.configure(checkConfig);
        check.setMessageDispatcher(dispatcher);

        final Exception exception = new IOException("test exception");
        Whitebox.invokeMethod(check, "logException", exception, new File(""));

        assertEquals("expected number of errors to fire", 1, dispatcher.savedErrors.size());
        final LocalizedMessage localizedMessage = new LocalizedMessage(1,
                Definitions.CHECKSTYLE_BUNDLE, "general.exception",
                new String[] {exception.getMessage()}, null, getClass(), null);
        assertEquals("Invalid message", localizedMessage.getMessage(),
                dispatcher.savedErrors.iterator().next().getMessage());
    }

